### PR TITLE
Feature/pytest

### PR DESCRIPTION
**Pull request by stopthatcow**
- [x] @aramnhammer
Add pytest, travis, python3 support, and coveralls

### CODE REVIEW

File: tests/test_style.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        bad_file_name = 'temp.c'
+        write_bad_file(bad_file_name)
+        ret = zazu.style.astyle([bad_file_name], {}, check=True, working_dir=dir)
+        assert bad_file_name in ret
+        ret = zazu.style.astyle([bad_file_name], {}, check=False, working_dir=dir)
+        assert bad_file_name in ret
+        ret = zazu.style.astyle([bad_file_name], {}, check=True, working_dir=dir)
+        assert not ret
+
+
```
Comment (aramnhammer) [inline tests/test_style.py]: extra line here and line 40 if you care.

File: tests/test_tool_helper.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    monkeypatch.setattr('zazu.tool.tool_helper.download_extract_tar_to_folder', lambda x, y, z: None)
+    for tool, versions in zazu.tool.tool_helper.get_tool_registry().iteritems():
+        for ver in versions:
+            enforcer = zazu.tool.tool_helper.get_enforcer(tool, ver)
+            assert not enforcer.check()
+            zazu.tool.tool_helper.install_spec('{}=={}'.format(tool, ver))
+            assert enforcer.check()
+            zazu.tool.tool_helper.uninstall_spec('{}=={}'.format(tool, ver))
+            assert not enforcer.check()
+
```
Comment (aramnhammer) [inline tests/test_tool_helper.py]: line here, and between tests in test_util and test_upgrade, if you care.

File: zazu/git_helper.py [hunk 3]
```diff
@@ -76,15 +77,24 @@ def install_git_hook(hooks_folder, hook_name, hook_resource_path):
         shutil.copy(hook_resource_path, hook_path)
 
 
+def parse_branch_return_list(branches):
+    ret = branches
+    for i, branch in enumerate(ret):
+        if branch.startswith('* '):
+            ret[i] = branch[2:]
+            break
+    return ret
```
Comment (aramnhammer) [inline zazu/git_helper.py]: We could use something like [inquirer](https://pypi.python.org/pypi/inquirer) to present a menu selection of branches. Possible idea for the future.

File: tests/test_style.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        write_bad_file(bad_file_name)
+        ret = zazu.style.autopep8([bad_file_name], {}, check=True, working_dir=dir)
+        assert bad_file_name in ret
+        ret = zazu.style.autopep8([bad_file_name], {}, check=False, working_dir=dir)
+        assert bad_file_name in ret
+        ret = zazu.style.autopep8([bad_file_name], {}, check=True, working_dir=dir)
+        assert not ret
+
+
+@pytest.fixture()
```
Comment (aramnhammer) [inline tests/test_style.py]: All fixtures can go into conftest.py.

File: tests/test_tool_helper.py (hunk 2 above)
Comment (stopthatcow) [inline tests/test_tool_helper.py]: fixed

File: zazu/git_helper.py (hunk 3 above)
Comment (stopthatcow) [inline zazu/git_helper.py]: yes, I also want to enable tab completion

**Comment by stopthatcow** [comment]
Ready for another once over @aramnhammer

**Pull request merged**